src/CSI.IBTA.BenefitsService/Services/ClaimsService.cs

```diff
@@ -0,0 +1,46 @@
+﻿using CSI.IBTA.BenefitsService.Interfaces;
+using CSI.IBTA.DataLayer.Interfaces;
+using CSI.IBTA.Shared.DTOs;
+using AutoMapper;
+using Microsoft.EntityFrameworkCore;
+using System.Reflection;
+using CSI.IBTA.Shared.Entities;
+
+namespace CSI.IBTA.BenefitsService.Services
+{
+    public class ClaimsService : IClaimsService
+    {
+        private readonly IBenefitsUnitOfWork _benefitsUnitOfWork;
+        private readonly IMapper _mapper;
+
+        public ClaimsService(IBenefitsUnitOfWork benefitsUnitOfWork, IMapper mapper)
+        {
+            _benefitsUnitOfWork = benefitsUnitOfWork;
+            _mapper = mapper;
+        }
+
+        public async Task<GenericResponse<PagedClaimsResponse>> GetClaims(int page, int pageSize, string claimNumber = "", string employerId = "", string claimStatus = "")
```

Comment:
you could make this more readable, for example:

public async Task<GenericResponse<PagedClaimsResponse>> GetClaims(int page, int pageSize, string claimNumber = "", string employerId = "", string claimStatus = "")
{
    var query = _benefitsUnitOfWork.Claims.GetSet()
        .Include(c => c.Plan.Package)
        .Include(c => c.Plan.PlanType);

    if (!string.IsNullOrEmpty(claimNumber))
    {
        query = query.Where(c => c.ClaimNumber.Contains(claimNumber));
    }

    if (!string.IsNullOrEmpty(employerId))
    {
        int employerIdInt;
        if (int.TryParse(employerId, out employerIdInt))
        {
            query = query.Where(c => c.Plan.Package.EmployerId == employerIdInt);
        }
    }

    if (!string.IsNullOrEmpty(claimStatus))
    {
        if (Enum.TryParse<ClaimStatus>(claimStatus, out ClaimStatus status))
        {
            query = query.Where(c => c.Status == status);
        }
    }

    var totalCount = await query.CountAsync();
    var totalPages = (int)Math.Ceiling((double)totalCount / pageSize);

    var claims = await query.OrderBy(p => p.Id)
        .Skip((page - 1) * pageSize)
        .Take(pageSize)
        .ToListAsync();

    var claimDtos = claims.Select(_mapper.Map<ClaimDto>).ToList();

    var response = new PagedClaimsResponse(claimDtos, page, pageSize, totalPages, totalCount);
    return new GenericResponse<PagedClaimsResponse>(null, response);
}